### ISSUE TITLE
Schema Integration

### ISSUE THREAD

**Issue by mkattnerBSB** [SignalK/artemis-server#1]
Hey guys, 

we are developing something pretty new for the marine sector. And we want to rely on signal k.

I just had a look to your source code. I like the way you did it.

But I couldn't find a full schema integrating. Are you doing it by hand, or ist there a automatic way to read it from the json files of the specification repository?
Do you have a roadmap?

Best,
Martin

**Comment by rob42** [comment]
Thanks, there are some improvements coming too, but I'm still finishing an edge case. The new version makes it easy to react to signalk keys and value changes, or to generate synthetic keys from real data - eg true wind from apparent wind.

'full schema integrating' - not sure what you mean. I'm deliberately using unstructured json (maps etc) as signalk both develops fast, and allows for arbitrary extension. Ive used jackson to reverse the schema to java pojos for the swagger docs, and tried the full schema too. Jackson didnt create nice pojos, so I canned that.

Roadmap is in my head!. 
Basically first goal is full signalk schema compliance with swagger docs for REST apis  etc. Multi-vessel, horizontally scale-able. Nearly there.
Second - fine-grained security (RBAC) down to the individual key level - the basics are there
Third - full history with analysis. - the basics are there
More:
- any transport ( MQTT, COAP, STOMP, serial, tcp, udp, etc)
- discovery and on-demand data sharing between arbitrary signalk  users - a bit like private chat groups in XMPP - in fact Ive used that to experiment.

But its a lot of work. Extra hands are always welcome.

**Comment by mkattner** [comment]
"...**swagger** docs, and tried the full schema too. Jackson didnt create nice **pojos**"

The signalk schema files are json schemas in version draft-04. Right? And no swagger docs.

As Jackson and the other POJO generators didn't work well, I wrote a simple own pojo generator.
Have a look to the attachment: 
![eclipse_2018-11-22_14-06-03](https://user-images.githubusercontent.com/6524345/48905961-4c0ea000-ee63-11e8-8e34-22d582942d76.png)

Deserializing with Jackosn:
`{"vessels":{"self":{"sensors":{"mastRotation":{"sensorData":"180"}}}}}`

Creates a nice SignalK Pojo Instance.

And serializing with Jackson results in:
`{"vessels":{"self":{"sensors":{"mastRotation":{"sensorData":"180"}}}}}`

Pojo Generation is 85% done and it works for my needs. I'll update it as soon as the first problems are araising.

If you want I can share the pojos with you ;-)
